packages/component/src/Composer.js

```diff
@@ -218,7 +219,11 @@ class Composer extends React.Component {
   }
 
   setLanguageFromProps(props) {
-    props.dispatch(setLanguage(props.locale || window.navigator.language || 'en-US'));
+    const lang = props.locale || window.navigator.language || 'en-US';
+    props.dispatch(setLanguage(lang));
+    if (['he', 'he-IL'].indexOf(lang) !== -1) {
+        props.dispatch(setDirection('rtl'));
```

Comment:
@compulim here I added the dispatch for set Direction since it is only relevant for specific languages. 
Now that I think of it this RTL languages array can be moved to a utility, with an isRtl() method.
But otherwise, you can dispatch the setDirection from anywhere in your code.